genesyscloud/knowledge_document_variation/genesyscloud_knowledge_document_variation_proxy.go

```diff
@@ -101,13 +105,62 @@ func (p *variationRequestProxy) GetAllKnowledgebaseEntities(ctx context.Context,
 	return p.GetAllKnowledgebaseEntitiesAttr(ctx, p, published)
 }
 
+// getVariationRequestByIdAndState reads the variation by ID and state
+// If no state is specified, get both published and draft variation and choose the most recent
+func (p *variationRequestProxy) getVariationRequestByIdAndState(ctx context.Context, ids *resourceIDs, state string) (*platformclientv2.Documentvariationresponse, *platformclientv2.APIResponse, error) {
+	if state == "" {
+		return p.getLatestPublishedOrDraftVariation(ctx, ids)
+	}
+	return p.getVariationRequestById(ctx, ids.knowledgeDocumentVariationID, ids.knowledgeDocumentID, ids.knowledgeBaseID, state, nil)
+}
+
+func (p *variationRequestProxy) getLatestPublishedOrDraftVariation(ctx context.Context, ids *resourceIDs) (*platformclientv2.Documentvariationresponse, *platformclientv2.APIResponse, error) {
+	return p.getLatestPublishedOrDraftVariationAttr(ctx, p, ids)
+}
+
+func getLatestPublishedOrDraftVariationFn(ctx context.Context, p *variationRequestProxy, ids *resourceIDs) (_ *platformclientv2.Documentvariationresponse, _ *platformclientv2.APIResponse, err error) {
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("getLatestPublishedOrDraftVariationFn: %w", err)
+		}
+	}()
+
+	time.Sleep(2 * time.Second) // wait for the variation to be published/draft
```

Comment:
Rather then just doing a flat out sleep.  Should this be a retry with multiple retries before it dies.

Reply:
done